filesystem/ImagickBackend.php

```diff
@@ -22,7 +22,25 @@ class ImagickBackend extends Imagick implements Image_Backend {
 	 */
 	public function __construct($filename = null) {
 		if(is_string($filename)) {
-			parent::__construct($filename);
+			try {
+				parent::__construct($filename);
+			} catch (Exception $e) {
+				if (file_exists($filename)) {
+					unlink($filename);
+				}
+				$filename = str_replace(Director::baseFolder() . '/', '', $filename);
```

Comment:
Removing the image record is risky; It could damage data records.

Also, making image resizing responsible for routing is extremely outside of the scope of this class. we should probably be catching errors at a much higher level.